packages/zed.rb

```diff
@@ -1,33 +1,54 @@
-require 'package'
+require 'buildsystems/rust'
 
-class Zed < Package
+class Zed < RUST
   description 'Zed is a high-performance, multiplayer code editor'
   homepage 'https://zed.dev/'
-  version '0.186.12'
+  version '0.196.6'
   license 'GPL-3, AGPL-3, Apache-2.0'
   compatibility 'x86_64'
-  min_glibc '2.28'
-  source_url "https://github.com/zed-industries/zed/releases/download/v#{version}/zed-linux-x86_64.tar.gz"
-  source_sha256 '9c0abca37319d96bf8ba03653a93413011210a338acc386ddaf88d68013c1a6d'
-
-  depends_on 'alsa_lib'
-  depends_on 'libbsd'
-  depends_on 'libxau'
-  depends_on 'libxcb'
-  depends_on 'libxdmcp'
-  depends_on 'libxkbcommon'
-  depends_on 'openssl'
-  depends_on 'zlib'
-  depends_on 'zstd'
-
-  no_compile_needed
-  no_shrink
+  source_url 'https://github.com/zed-industries/zed.git'
+  git_hashtag "v#{version}"
+  binary_compression 'tar.zst'
+
+  binary_sha256({
+     x86_64: 'fd786575ec2c8e35ecee5e71164d053d77b65f85c190c051b1925fb8d7585212'
+  })
+
+  depends_on 'alsa_lib' # R
+  depends_on 'cargo_about' => :build
+  depends_on 'gcc_lib' # R
+  depends_on 'glibc' # R
+  depends_on 'libbsd' # R
+  depends_on 'libx11' # R
+  depends_on 'libxau' # R
+  depends_on 'libxcb' # R
+  depends_on 'libxdmcp' # R
+  depends_on 'libxkbcommon' # R
+  depends_on 'llvm_dev' => :build
+  depends_on 'openssl' # R
+  depends_on 'ruby_solargraph' # L
+  depends_on 'rust' => :build
+  depends_on 'zlib' # R
+  depends_on 'zstd' # R
+
+  rust_flags '-C link-args=-Wl,--disable-new-dtags,-rpath,\$ORIGIN/../lib -C symbol-mangling-version=v0 --cfg tokio_unstable'
```

Comment:
I don't think you need to escape the `$` in single quotes?

Reply:
Done.